src/Header/Cookie.php

```diff
@@ -97,13 +97,30 @@ public function getFieldValue()
     {
         $nvPairs = [];
 
-        foreach ($this as $name => $value) {
+        $cookies = [];
+        foreach ($this->flattenCookies($cookies, $this) as $name => $value) {
             $nvPairs[] = $name . '=' . (($this->encodeValue) ? urlencode($value) : $value);
         }
 
         return implode('; ', $nvPairs);
     }
 
+    protected function flattenCookies(&$result, $data, $prefix = null)
```

Comment:
For me the correct way seems to be what php is doing. So yes we need a recursive function. But you are right we can write it without passing by reference.